### PR TITLE
Add composite action 'molecule-test'

### PR DESCRIPTION
**Pull request by p-j-smith**
- add a composite action that can be used for running `molecule test` in standalone Ansible roles
- takes a single input `scenario` that determines which scenario to run; defaults to `default`

### CODE REVIEW

File: actions/molecule-test/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+This action can be used in the following manner:
+
+```yaml
+jobs:
+  molecule:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run `molecule test`
+        uses: UCL-MIRSG/.github/actions/molecule-test@v.0.1.0
```
Comment (drmatthews) [inline actions/molecule-test/README.md:11]: Should the ref be `v0.1.0`? And should we keep the version stated in `README.md` up-to-date with the repo tag (currently `v0.21.0`)?

File: actions/molecule-test/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        molecule_scenario:
+          - centos7
+          - rocky8
+    steps:
+      - name: Run `molecule test`
+        uses: UCL-MIRSG/.github/actions/molecule-test@v.0.1.0
```
Comment (drmatthews) [inline actions/molecule-test/README.md:29]:
```suggestion
        uses: UCL-MIRSG/.github/actions/molecule-test@v0.21.0
```

**Review by drmatthews — CHANGES_REQUESTED** [review]
This is a great idea!

File: actions/molecule-test/README.md (hunk 1 above)
Comment (p-j-smith) [reply actions/molecule-test/README.md:11]: I think it's not really possible to keep the README up to date with the repo tag, as a new release is automatically made every time a pr is merged (and I guess that would mean we would need to update the README on every pr). Unless there's a way to automatically keep it up to date?

**Comment by p-j-smith** [comment]
> This is a great idea!

Inspired by @paddyroddy 's work!

File: actions/molecule-test/README.md (hunk 1 above)
Comment (drmatthews) [reply actions/molecule-test/README.md:11]: Sorry - I didn't see the workflow for auto-tagging. In that case maybe the README should just say "vx.x.x" to indicte that a version should go there? Sorry for pendantry - I'll approve and you can choose to ignore if you like.

**Review by drmatthews — APPROVED** [review]
👍

File: actions/molecule-test/README.md (hunk 1 above)
Comment (p-j-smith) [reply actions/molecule-test/README.md]: yeah that's probably a neater way of doing it, I've updated it now. Thanks for approving!

Comment (paddyroddy) [reply actions/molecule-test/README.md]: We should probably have it consistent throughout the `README.md`s. The problem is that there isn't a convenient `pre-commit autoupdate` equivalent for actions.

**Comment by paddyroddy** [comment]
If this is something that will be used generally, can you add it to the template repo too @p-j-smith ?

File: actions/molecule-test/README.md (hunk 1 above)
Comment (p-j-smith) [reply actions/molecule-test/README.md]: yeah consistency would be good, are you happy with the `vx.y.z` convention used in this action?

**Comment by p-j-smith** [comment]
> If this is something that will be used generally, can you add it to the template repo too @p-j-smith ?

It will be used for any ansible role (and maybe collection too?). But I think it would be better in the [ansible-role-template](https://github.com/UCL-MIRSG/ansible-role-template) that @drmatthews created, as we wouldn't use it for any other repos. We should also add the other actions etc. to that role template repo

File: actions/molecule-test/README.md (hunk 1 above)
Comment (paddyroddy) [reply actions/molecule-test/README.md]: Yeah, just raising a PR as we speak

**Comment by paddyroddy** [comment]
> It will be used for any ansible role (and maybe collection too?). But I think it would be better in the [ansible-role-template](https://github.com/UCL-MIRSG/ansible-role-template) that @drmatthews created, as we wouldn't use it for any other repos. We should also add the other actions etc. to that role template repo

Coolio

**Pull request merged**